### PR TITLE
Add endpoint detection for init outside the document

### PR DESCRIPTION
**Pull request by NipunaMarcus**
## Purpose
  - Add endpoint detection for init outside the document
## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: misc/diagram-util/src/main/java/org/ballerinalang/diagramutil/SyntaxTreeMapGenerator.java [hunk 1]
```diff
@@ -255,13 +283,52 @@ private JsonObject visibleEP(Node node, TypeSymbol typeSymbol) {
...
         return symbolMetaInfo;
     }
 
+    private boolean isAvailableAsEndpoint(String name) {
+        boolean isAvailable = false;
+
+        for (JsonObject ep : this.visibleEpsForEachBlock) {
+            isAvailable = ep.get("name").getAsString().equals(name);
+            if (isAvailable) {
+                break;
```
Comment (prabushi) [inline misc/diagram-util/src/main/java/org/ballerinalang/diagramutil/SyntaxTreeMapGenerator.java:316]: Can we return if available?

Comment (NipunaMarcus) [reply misc/diagram-util/src/main/java/org/ballerinalang/diagramutil/SyntaxTreeMapGenerator.java]: Fixed

File: misc/diagram-util/src/main/java/org/ballerinalang/diagramutil/SyntaxTreeMapGenerator.java [hunk 2]
```diff
@@ -176,6 +177,15 @@ protected JsonElement transformSyntaxNode(Node node) {
                 // Ignore as semantic API calls cannot break the ST JSON creation.
             }
 
+            if (node.kind() == SyntaxKind.REMOTE_METHOD_CALL_ACTION) {
+                RemoteMethodCallActionNode remoteMethodCallActionNode = (RemoteMethodCallActionNode) node;
+                Optional<Symbol> expressionSymbol = this.semanticModel.symbol(remoteMethodCallActionNode.expression());
+                if (expressionSymbol.isPresent()) {
```
Comment (nadeeshaan) [inline misc/diagram-util/src/main/java/org/ballerinalang/diagramutil/SyntaxTreeMapGenerator.java:183]: Please check for the symbol kind before casting, for safety.

**Review by nadeeshaan — APPROVED** [review]

File: misc/diagram-util/src/main/java/org/ballerinalang/diagramutil/SyntaxTreeMapGenerator.java (hunk 2 above)
Comment (NipunaMarcus) [reply misc/diagram-util/src/main/java/org/ballerinalang/diagramutil/SyntaxTreeMapGenerator.java]: fixed

**Pull request merged**